src/chains/ethereum/ethereum/src/helpers/precompiles.ts

```diff
@@ -19,14 +16,19 @@ const PRECOMPILED_ACCOUNT: Account = {
   serialize: () => SERIALIZED_PRECOMPILE
 } as any;
 
-const accountCache: Address[] = [];
-const makeAccount = (i: number): Address => {
-  if (accountCache[i]) return accountCache[i];
+const accountCache: Map<number, Address> = new Map();
+const getOrCreateAccount = (i: number): Address => {
+  if (accountCache.has(i)) return accountCache.get(i);
```

Comment:
Not so fun fact, internally to v8 Map's `.has` is a `.get`, but with an additional check to see if it existed. So `get`ing and then checking it is always going to be about 2x faster than checking via `.has` and then returning via `get`.

Since the iterations here is so very small the difference likely isn't measurable in most or all real workflows, so the change might not be worth while.... BUUUUTTTT since the more performant alternative isn't too hard to read I wouldn't hate it if it replaced the has/get pattern (even though the has/get is so nice to read!):

```
  let cachedAddress = accountCache.get(i);
  if (cachedAddress) return cachedAddress;
```

Ultimately up to you. Just thought I'd mention it!

Reply:
That's great to know, thanks! I'll fix this.